src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/BackticksUnquoter.java

```diff
@@ -0,0 +1,62 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.utils;
+
+/**
+ * Utils Class for remove the back-ticks of identifiers
+ */
+public class BackticksUnquoter {
+
+    /**
+     *
+     * @param text
+     * @return An unquoted string whose outer pair of back-ticks (if any) has been removed
+     */
+    public static String unquoteSingleField(String text) {
+        if (text != null && text.startsWith("`") && text.endsWith("`")) {
```

Comment:
extract this into helper

isQuoted(string)

and then you can reuse it in UnquoteIdentifierRule

Reply:
@galkk yes this would be better to extract isQuoted, to make the code cleaner and reusable. it's done now, thanks!